diff --git internal/generationtest/generationtest.bzl internal/generationtest/generationtest.bzl
index 396a59a..1d7760f 100644
--- internal/generationtest/generationtest.bzl
+++ internal/generationtest/generationtest.bzl
@@ -4,7 +4,7 @@ Test for generating rules from gazelle.
 
 load("@io_bazel_rules_go//go:def.bzl", "go_test")
 
-def gazelle_generation_test(name, gazelle_binary, test_data, build_in_suffix = ".in", build_out_suffix = ".out"):
+def gazelle_generation_test(name, gazelle_binary, test_data, build_in_suffix = ".in", build_out_suffix = ".out", **kwargs):
     """
     gazelle_generation_test is a macro for testing gazelle against workspaces.
 
@@ -51,4 +51,5 @@ def gazelle_generation_test(name, gazelle_binary, test_data, build_in_suffix = "
         data = test_data + [
             gazelle_binary,
         ],
+        **kwargs
     )
