diff --git cmd/gazelle/fix-update.go cmd/gazelle/fix-update.go
index 47d00b3..7285cda 100644
--- cmd/gazelle/fix-update.go
+++ cmd/gazelle/fix-update.go
@@ -387,6 +387,12 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 			unionKindInfoMaps(kinds, v.mappedKindInfo))
 	}
 
+	for _, lang := range languages {
+		if finishable, ok := lang.(language.ResolveFinishableLanguage); ok {
+			finishable.DoneResolving(c)
+		}
+	}
+
 	// Emit merged files.
 	var exit error
 	for _, v := range visits {
diff --git language/lang.go language/lang.go
index 4f3d7e1..2befb21 100644
--- language/lang.go
+++ language/lang.go
@@ -108,6 +108,10 @@ type FinishableLanguage interface {
 	DoneGeneratingRules()
 }
 
+type ResolveFinishableLanguage interface {
+	DoneResolving(c *config.Config)
+}
+
 // GenerateArgs contains arguments for language.GenerateRules. Arguments are
 // passed in a struct value so that new fields may be added in the future
 // without breaking existing implementations.
